core/app/models/spree/promotion/rules/taxon.rb

```diff
@@ -5,30 +5,43 @@ class Taxon < PromotionRule
         has_many :promotion_rule_taxons, class_name: 'Spree::PromotionRuleTaxon', foreign_key: :promotion_rule_id
         has_many :taxons, through: :promotion_rule_taxons, class_name: 'Spree::Taxon'
 
-        MATCH_POLICIES = %w(any all)
+        MATCH_POLICIES = %w(any all none)
         preference :match_policy, default: MATCH_POLICIES.first
 
         def applicable?(promotable)
           promotable.is_a?(Spree::Order)
         end
 
         def eligible?(order, _options = {})
+          order_taxons = taxons_in_order_including_parents(order)
           if preferred_match_policy == 'all'
-            unless (taxons.to_a - taxons_in_order_including_parents(order)).empty?
+            unless (taxons.to_a - order_taxons).empty?
               eligibility_errors.add(:base, eligibility_error_message(:missing_taxon))
             end
-          else
-            order_taxons = taxons_in_order_including_parents(order)
+          elsif preferred_match_policy == 'any'
             unless taxons.any?{ |taxon| order_taxons.include? taxon }
               eligibility_errors.add(:base, eligibility_error_message(:no_matching_taxons))
             end
+          elsif preferred_match_policy == 'none'
+            unless taxons.none?{ |taxon| order_taxons.include? taxon }
+              eligibility_errors.add(:base, eligibility_error_message(:has_excluded_taxon))
+            end
+          else
+            raise "unexpected match policy: #{preferred_match_policy.inspect}"
```

Comment:
I want to fix this, realized in production that a raise here can keep anyone from checking out ever, which is bad. 

Previously, with an unrecognized match_policy it just assumed it meant `any`.  I think the alternatives are keeping that behavior, or just rejecting eligibility (and logging?) with an unrecognized `match_policy`. 

Any thoughts?